pages/agent/cli_pipeline.md.erb

```diff
@@ -42,13 +42,64 @@ Options:
    --debug-http                                 Enable HTTP debug mode, which dumps all request and response bodies to the log [$BUILDKITE_AGENT_DEBUG_HTTP]
 ```
 
-## Pipeline Format
+## Pipeline format
 
 The pipeline can be written as YAML or JSON, but YAML is more commonly for its readability. There are two top level properties you can specify:
 
-* `env` - a hash of build-wide environment variables
-* `steps` - an array of build steps to be run (a mix of script, [`wait`](/docs/pipelines/wait-step) and [`block`](/docs/pipelines/block-step) steps)
+* `env` - A map of <a href="/docs/builds/environment-variables">environment variables</a> to apply to all steps
+* `steps` - An list of [build pipeline steps](/docs/pipelines/defining-steps)
 
+## Environment variable substitution
 
+The `pipeline upload` command in Buildkite Agent versions 3.0 and above supports environment variable substitution using the syntax `$VAR` and `${VAR}`.
 
+For example, the following pipeline substitutes a number of [Buildkite’s default environment variables](/docs/builds/environment-variables) into a [trigger step](/docs/pipelines/trigger-step):
 
+```yml
+- command: "$BUILDKITE_PIPELINE_SLUG-deploy"
+  label: "\:rocket\:"
+  branches: "master"
+  async: true
+  build:
+    message: "$BUILDKITE_MESSAGE"
+    commit: "$BUILDKITE_COMMIT"
+    branch: "$BUILDKITE_BRANCH"
+```
+
+### Default values
+
+If an environment variable has not been set its default value will be a blank string. You can set your own default value using the syntax `${VAR:-default}`.
+
+For example, the following step will run the command `deploy.sh staging` if the `DEPLOY_ENV` environment variable has not been set or is blank (i.e. `""`):
+
+```yaml
+- command: "deploy.sh \"${DEPLOY_ENV:-staging}\""
+```
+
+If you need to substitute blank values (i.e. `""`) you can use the syntax `${VAR-default}`.
+
+For example, if the `DEPLOY_ENV` environment variable is set to `""` the following step will run the command `deploy.sh ""`, and if it has not been set it will run the command `deploy.sh "staging"`:
+
+```yaml
+- command: "deploy.sh \"${DEPLOY_ENV-staging}\""
+```
+
+### Character ranges
+
+You can substitute a range of characters from an environment variable by specifying a start and end position using the syntax `${VAR:start:end}`.
+
+For example, the following step will echo the first 7 characters of the `BUILDKITE_COMMIT` environment variable:
+
+```yaml
+- command: "echo ${BUILDKITE_COMMIT:0:7}"
+```
+
+If the environment variable has not been set, the range will return a blank string.
+
+### Buildkite Agent 2.x support
+
+If you want to use environment variable substitution, but are unable to update your Buildkite Agent to version 3.0 or above, it is possible (but not recommended) to emulate the behavior using your shell’s built-in environment variable substitution:
```

Comment:
I think they probably already want to use env vars if they've made it this far down the page 😄 
We could start this one with the 'if you're unable to update your agent to 3.0' bit. It's a long sentence at the moment with a lot of different things in there, you end up a little lost by the end.